File: java/app/src/main/java/org/example/naver/NaverClicker.java
package org.example.naver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

import java.util.ArrayList;
import java.util.List;

public class NaverClicker {
    public static void run() throws InterruptedException {
        // 1️⃣ ChromeDriver 자동 설치
        WebDriverManager.chromedriver().setup();

        // 2️⃣ 브라우저 실행
        WebDriver driver = new ChromeDriver();
        driver.manage().window().maximize();

        try {
            // 3️⃣ naver.com 접속
            driver.get("https://www.naver.com");
            Thread.sleep(2000); // 페이지 로딩 대기

            // 4️⃣ 클릭 전 HTML 가져오기
            String htmlBefore = driver.getPageSource();
            System.out.println("클릭 전 HTML 길이: " + htmlBefore.length());

            // 5️⃣ 모든 버튼 클릭
            List<WebElement> buttons = driver.findElements(By.tagName("button"));
            for (WebElement btn : buttons) {
                try {
                    btn.click();
                } catch (Exception e) {
                    // 클릭 불가 버튼 무시
                }
            }

            Thread.sleep(2000); // 클릭 후 대기

            // 6️⃣ 클릭 후 HTML 가져오기
            String htmlAfter = driver.getPageSource();
            System.out.println("클릭 후 HTML 길이: " + htmlAfter.length());

            // 7️⃣ 버튼 텍스트 가져오기
            buttons = driver.findElements(By.tagName("button"));
            List<String> buttonTexts = new ArrayList<>();
            for (WebElement btn : buttons) {
                buttonTexts.add(btn.getText());
            }

            // 특정 키워드 포함 버튼 필터링
            String keyword = "확인";
            List<String> matchedButtons = new ArrayList<>();
            for (String text : buttonTexts) {
                if (text.contains(keyword)) {
                    matchedButtons.add(text);
                }
            }

            System.out.println("찾은 버튼들: " + matchedButtons);

        } finally {
            // 8️⃣ 브라우저 종료
            driver.quit();
        }
    }
}

File: kotlin/app/src/main/kotlin/org/example/naver/NaverClicker.kt
package org.example

import io.github.bonigarcia.wdm.WebDriverManager
import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.chrome.ChromeDriver

object NaverClicker {
    @JvmStatic
    fun run() {
        WebDriverManager.chromedriver().setup()
        val driver: WebDriver = ChromeDriver()
        driver.manage().window().maximize()

        try {
            driver.get("https://www.naver.com")
            Thread.sleep(2000)

            val htmlBefore = driver.pageSource
            println("클릭 전 HTML 길이: ${htmlBefore.length}")

            val buttons: List<WebElement> = driver.findElements(By.tagName("button"))
            buttons.forEach {
                try { it.click() } catch (_: Exception) {}
            }

            Thread.sleep(2000)

            val htmlAfter = driver.pageSource
            println("클릭 후 HTML 길이: ${htmlAfter.length}")

            val buttonTexts = driver.findElements(By.tagName("button")).map { it.text }
            val keyword = "확인"
            val matchedButtons = buttonTexts.filter { it.contains(keyword) }

            println("찾은 버튼들: $matchedButtons")
        } finally {
            driver.quit()
        }
    }
}
